independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/ArcTestContainer.java

```diff
@@ -248,6 +254,7 @@ public ArcTestContainer build() {
     private final List<BuildCompatibleExtension> buildCompatibleExtensions;
 
     private final boolean strictCompatibility;
+    private final boolean optimizeContexts;
```

Comment:
Why don't we default to `true` here?

Reply:
Well, I think that it's a kind of an advanced feature. So it makes sense to enable it by default for Quarkus but not in general. Also we do the same for `ArcTestContainer.removeUnusedBeans`...

Reply:
Fair enough. It seems to me that we should just force-enable this in ArC tests, without even an option to go back, but I won't insist :-)